src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,25 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_properties_from_model_in_path(output_model, input_model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    output_document = {"properties": output_model.copy()}
+    input_document = {"properties": input_model.copy()}
+    for path in paths:
+        try:
+            _prop, _resolved_path, _parent = traverse(input_document, path)
```

Comment:
To make this better understanding, could we wrap the path check into a new method? e.g. 

```
def path_exists(input_doc, path) 
 try:
            _prop, _resolved_path, _parent = traverse(input_document, path)
 except LookupError:
    return False
return True

''' Use it in the prune
for path in paths
  if not path_exists(input_document, path)
    ''' Here we can also make a method to say delete_path
     _prop, resolved_path, parent = traverse(output_document, path)
     key = resolved_path[-1]
     del parent[key]
```

Reply:
sure, will do